fawltydeps/main.py

```diff
@@ -290,6 +292,13 @@ def main(
         analysis = Analysis.create(settings, stdin)
     except UnparseablePathException as exc:
         return parser.error(exc.msg)  # exit code 2
+    except UnresolvedDependenciesError as exc:
+        logger.error(
+            "%s \nIf you use '--install-deps' option "
+            "please make sure that you have all your dependencies installed.",
+            str(exc.msg),
+        )
+        return 5
```

Comment:
Agree that this should be solved in another issue.

It seems that we might need to think a bit more about how to best handle errors from `pip install`. Here's what I found after digging for a little bit:
- We currently try to install _all_ of `requirements` with a single `pip install` command. This is ~necessary for performance reasons, I believe (to prevent running one subprocess per requirement/dependency).
- If the `pip install` fails, we currently don't know whether that means _nothing_ was installed, or if _some_ of the requirements might have been installed (but others failed).
- I can think of scenarios where _nothing_ would be installed (e.g. network issues) and other scenarios where almost all deps were installed (e.g. one of the deps doesn't exist due to e.g. a typo).
- We _could_ play games with `pip` to try to figure out which dependencies were or were not installed.
  - e.g. using the new `--report` option to get a [detailed JSON report](https://pip.pypa.io/en/stable/reference/installation-report/), or
  - using `pip inspect` afterwards to get a similar JSON report, or
  - using the simpler (and backwards-compatible) `pip list` to get a list of installed dependencies.
- _However_, we already rely on `LocalPackageResolver` to look at the virtualenv and return the installed packages for us, so we're probably better off ignoring any `pip install` failure and leave `LocalPackageResolver` to tell us which packages were installed or not.
- Maybe the safest approach is something like this?:
  1. First try to install all dependencies with a single `pip install` command. If this succeeds, then skip directly to step 3.
  2. If the above fails, fall back to `pip install` one dependency at a time. (This might be very slow, but will hopefully be skipped most of the time.)
  3. Use the existing `LocalPackageResolver` code to resolve what can be resolved, and leave the rest as unresolved.

Reply:
created https://github.com/tweag/FawltyDeps/issues/322